### PR TITLE
Fix VarHandle return bug.

### PR DESCRIPTION
**Pull request by gongweibao**
Fix https://github.com/PaddlePaddle/Paddle/pull/13307#pullrequestreview-154461070

### CODE REVIEW

File: paddle/fluid/operators/distributed/request_handler.h [hunk 1]
```diff
@@ -68,12 +68,14 @@ class VarHandle {
 
  public:
   bool Wait() {
+    int ret = kVarHandleDefaultState;
```
Comment (typhoonzero) [inline paddle/fluid/operators/distributed/request_handler.h]: why not use bool?

File: paddle/fluid/operators/distributed/request_handler.h [hunk 2]
```diff
@@ -68,12 +68,14 @@ class VarHandle {
...
     {
       std::unique_lock<std::mutex> lk(sync_mutex_);
       wait_cond_.wait(lk, [this] { return ok_ != kVarHandleDefaultState; });
+      ret = ok_;
```
Comment (typhoonzero) [inline paddle/fluid/operators/distributed/request_handler.h]: wait is a blocking call, why need a return value?

File: paddle/fluid/operators/distributed/request_handler.h (hunk 1 above)
Comment (gongweibao) [reply paddle/fluid/operators/distributed/request_handler.h]: 你说的对，是我理解错了！

File: paddle/fluid/operators/distributed/request_handler.h (hunk 2 above)
Comment (gongweibao) [reply paddle/fluid/operators/distributed/request_handler.h]: Done. Thanks!

**Pull request merged**